Incubation announce prep (#101)
fluxcd/website: content/en/blog/2021-03-08-flux-is-a-cncf-incubation-project.md

---
author: dholbach
date: 2021-03-10 06:30:00+00:00
title: Flux is a CNCF Incubation project
description: The Cloud Native Computing Foundation promoted Flux to Incubation status. We celebrate all the great work that turned Flux into a GitOps family of projects.
url: /blog/2021/03/flux-is-a-cncf-incubation-project/
---

![Incubation](/img/incubation.png)

The Flux community is proud to announce that the CNCF Technical
Oversight Committee just promoted the Flux project to Incubating status!
🎉

For us as a project, this is proof not only of the wide-spread use of
Flux (and Helm Operator) v1, but also of our big strides into becoming a
GitOps family of projects and bringing a toolkit approach to Continuous
and Progressive Delivery.

We are particularly happy about the support from our community, the
project maintainers, contributors and end-users who helped us through
this process. Thanks to our friends at Deutsche Telekom, Lunar,
MediaMarktSaturn and Sortlist who spoke in favour of Flux and explained
their GitOps implementation in user interviews with the CNCF.

> ![Incubation Infographics](/img/incubation-infographics.png)

Reviewer: Can we center this image? Here's an inline suggestion below

```suggestion
![Incubation Infographics](/img/incubation-infographics.png#center)
```

which - if it still works - should just need a CSS addition: https://discourse.gohugo.io/t/handling-images-size-aligning/1940/2

Author: I'm off for today and will be off tomorrow too. No time to check CSS additions. :man_shrugging:  Seperate PR?

> "We created Flux as an open source project because we believe that is
> the best environment in which to develop software. Since then I have
> continually been shown that trusting your users and contributors is
> rewarded. Although it is still a modest project, there is a loyal and
> growing community around Flux and the methodology it engendered,
> GitOps. I see the acceptance of Flux into CNCF incubation status as
> validation of that community as much as the software itself."
>
> \-- Michael Bridgen, Flux co-creator

## What's next for Flux

For 10 months we have been working on the new iteration of Flux. It is
based on modern tooling, very composable and we were able to add
long-requested features in next to no time.

Flux as a project has been on this trajectory for a longer time already:
Flux was started in 2016, the Helm Operator was added in 2018, kustomize
support added in 2019. 2020 was the year in which re-started the project
to turn Flux into a GitOps family of projects, where simple and focused
controllers can naturally be composed. 2021 saw Flagger being moved into
the Flux organization and will see the GA release of Flux v2 in the
coming months.

> ## Flux v2 is getting closer

Reviewer: Somehow "Flux 2 is getting closer" sounds like it's not ready to use? GA helps clarify what we're getting closer to. WDYT?

```suggestion
## Flux v2 getting closer to GA
```

<img alt="Flux v2 work" style="background-color:#1b8de2; width: 55vw; min-width: 330px;" src="/img/building-blocks.svg" />

We are very proud of where Flux v2 is at now, and are carefully working
toward a GA release, at which point we'll recommend that all Flux v1
users migrate to Flux v2.

Now is the perfect time to familiarise yourself with [Flux
v2](https://fluxcd.io) - the Get Started guide only takes a

> couple of minutes to complete. If you would prefer a video, check out

Reviewer: Suggestion from @kingdonb:
```suggestion
couple of minutes to complete. If you prefer a video, check out
```

[our resources section](https://fluxcd.io/#resources).

Migrating from v1 will require some work, but it will definitely be
worth it: in this iteration you are going to get

> - Multiple Git repositories, multi tenancy
> - Better performance
> - Way more flexible, modern tooling
> - And lots of other new features

Reviewer: Change suggestion from @stefanprodan:

```suggestion
- Support for multiple Git repositories
- Operational insight through health checks, events and alerts
- Multi-tenancy capabilities
- Better performance
- And lots of other new features
```

We've taken our time to ensure that Flux v2 has feature parity with Flux
v1, and that end users have the best experience possible migrating to
Flux v2. Stay tuned for our upcoming announcement about the Flux v2
migration and support timeline.

## Get started today

If you have been waiting to start your GitOps journey, today is a good
time to start!

> Check out the following upcoming events

Reviewer: ```suggestion
Check out these upcoming events:
```

> 8 Mar 2021 - [Migrating from Flux v1 to Flux v2 with Leigh
> Capili](https://www.meetup.com/GitOps-Community/events/276539791/)
>
> 22 Mar 2021 - [Hands-On GitOps Patterns for Helm Users with Scott
> Rigby](https://weaveworks.zoom.us/webinar/register/WN_rpXk5yhYQN2zxIRUNDxCow)
>
> 25 Mar 2021 - CNCF webinar: Flux is Incubating + The Road Ahead with
> Stefan Prodan & Leigh Capili
>
> 5 Apr 2021 - [Flux v2 on Azure with Leigh
> Capili](https://www.meetup.com/GitOps-Community/events/276674768/)
>
> 19 Apr 2021 - [Setting up Notifications, Alerts, & Webhook with Flux
> v2 by Alison Dowdney](https://www.meetup.com/GitOps-Community/events/276582835/)

Or dive straight into

> [https://fluxcd.io/](https://toolkit.fluxcd.io/get-started/)

Reviewer 2: this link might be super confusing, as it reads as fluxcd.io but points to the get started guide

```suggestion
our [Get Started guide](https://toolkit.fluxcd.io/get-started/)
```

Author: Thanks for catching this issue... your change WFM, Tamao suggested we just point to fluxcd.io though...?

@mewzherder WDYT?
